bundles/framework/layerselector2/Flyout.js

```diff
@@ -588,19 +601,26 @@ Oskari.clazz.define('Oskari.mapframework.bundle.layerselector2.Flyout',
         deactivateAllFilters: function(notDeactivateThisFilter){
             var me = this;
 
-            jQuery.each(me.filters, function(index, filter) {
-                if(!notDeactivateThisFilter || filter.filterName !== notDeactivateThisFilter) {
-                    var filterIcon = jQuery('.filter-icon.' + 'filter-'+filter.filterName);
-                    filterIcon.removeClass(filter.iconClassActive);
-                    filterIcon.removeClass('active');
-                    filterIcon.addClass(filter.iconClassDeactive);
-                    filterIcon.parents('.filter').attr('title',filter.tooltip);
-                }
+            me._currentFilter = null;
+            me.layerTabs.forEach(function(tab, tabIndex){
+                var filterContainer = tab.getTabPanel().getContainer().find('.layerselector2-layer-filter');
+                jQuery.each(me.filters, function(index, filter) {
+                    if(!notDeactivateThisFilter || filter.filterName !== notDeactivateThisFilter) {
+                        var filterIcon = filterContainer.find('.filter-icon.' + 'filter-'+filter.filterName);
+                        filterIcon.removeClass(filter.iconClassActive);
+                        filterIcon.removeClass('active');
+                        filterIcon.addClass(filter.iconClassDeactive);
+                        // Set tooltip for one per filter
+                        if(tabIndex === 0) {
```

Comment:
Only set tooltips for the first tab?

Reply:
_setFilterTooltip-function already adds tooltips for all tabs so there needs call this function once time per filter.